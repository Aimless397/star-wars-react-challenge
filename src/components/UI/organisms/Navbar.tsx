import "../../../styles/app.css";

export const Navbar = () => {
  return (
    <>
      <nav className="navbar">
        <span className="navbar-title text-light">Ravn Star Wars Registry</span>
      </nav>
    </>
  );
};
