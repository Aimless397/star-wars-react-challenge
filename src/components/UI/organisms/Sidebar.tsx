import { useContext } from "react";
import "../../../styles/sidebar.css";
import { LoadingCell } from "../atoms/LoadingCell";
import { NoticeCell } from "../atoms/NoticeCell";
import { PersonCell } from "../molecules/PersonCell";
import { CharacterContext } from "../../../context/CharacterContext";

export const Sidebar = () => {
  const { characters, isLoading, hasError, nextPage } =
    useContext(CharacterContext);

  return (
    <div className="col-5 col-sm-4 col-md-4 col-lg-4 col-xl-3 col-xxl-3">
      <div className="sidebar">
        {!hasError && <PersonCell data={characters ?? []} />}

        {isLoading ? (
          <LoadingCell />
        ) : (
          !hasError && (
            <div className="d-flex">
              <button
                className="btn btn-secondary my-2 mx-auto"
                onClick={() => nextPage()}
              >
                Load more Characters
              </button>
            </div>
          )
        )}
        {!isLoading && hasError && <NoticeCell />}
      </div>
    </div>
  );
};
