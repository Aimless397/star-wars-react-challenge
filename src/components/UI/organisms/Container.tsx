import { Sidebar } from "./Sidebar";
import { CharacterContent } from "../templates/CharacterContent";

export const Container = () => {
  return (
    <div data-testid="contentContainer" className="row m-0 contentContainer">
      <Sidebar />
      <CharacterContent />
    </div>
  );
};
