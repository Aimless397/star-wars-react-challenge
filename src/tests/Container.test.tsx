import { describe, expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import { Container } from "../components/UI/organisms/Container";

describe("<Container />", () => {
  test("should be rendered", () => {
    const { container } = render(<Container />);

    expect(container).toMatchSnapshot();
  });

  test("should have the correct className style", () => {
    render(<Container />);

    const divElement = screen.getByTestId("contentContainer");

    expect(divElement.className).toContain("contentContainer");
  });
});
