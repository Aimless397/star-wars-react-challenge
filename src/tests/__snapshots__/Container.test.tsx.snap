// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should be rendered 1`] = `
<div>
  <div
    class="row m-0 contentContainer"
    data-testid="contentContainer"
  >
    <div
      class="col-5 col-sm-4 col-md-4 col-lg-4 col-xl-3 col-xxl-3"
    >
      <div
        class="sidebar"
      >
        <div
          class="d-flex"
        >
          <button
            class="btn btn-secondary my-2 mx-auto"
          >
            Load more Characters
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-7 col-sm-8 col-md-8 col-lg-8 col-xl-9 col-xxl-9"
    >
      <div
        class="row m-0"
      >
        <div
          class="col"
        />
        <div
          class="col-12 col-sm-10 col-md-9 col-lg-9 col-xl-9 col-xxl-8"
        >
          <div
            class="charactersContainer"
          >
            <div
              class="sectionContainer"
            >
              <span
                class="sectionHeader"
                data-testid="sectionHeader"
              >
                General Information
              </span>
            </div>
            <div
              class="dataCellContainer"
            >
              <span
                class="dataCellCategoryText"
                data-testid="dataCellCategoryText"
              >
                Eye Color
              </span>
            </div>
            <div
              class="dataCellContainer"
            >
              <span
                class="dataCellCategoryText"
                data-testid="dataCellCategoryText"
              >
                Hair Color
              </span>
            </div>
            <div
              class="dataCellContainer"
            >
              <span
                class="dataCellCategoryText"
                data-testid="dataCellCategoryText"
              >
                Skin Color
              </span>
            </div>
            <div
              class="dataCellContainer"
            >
              <span
                class="dataCellCategoryText"
                data-testid="dataCellCategoryText"
              >
                Birth Year
              </span>
            </div>
            <div
              class="sectionContainer"
            >
              <span
                class="sectionHeader"
                data-testid="sectionHeader"
              >
                Vehicles
              </span>
            </div>
          </div>
        </div>
        <div
          class="col"
        />
      </div>
    </div>
  </div>
</div>
`;
